### DISCUSSION TITLE
Screen Responsiveness.... Is there a fixed base function on Pixi.js?

### DISCUSSION THREAD

**Discussion by inglesuniversal** [pixijs/pixijs#7225] [General]
I have tried pixi.js on and off and even though it is an awesome library, I always find it hard to get started due to a single issue.

Most tutorials start by showing how do draw a small square with an image inside a black canvas.

Is there a prebuilt function I can use from the start that can make my canvas fill the entire screen and keep the content in place and in the same area of the screen regardless if I turn my, lets say (iphone or ipad) from portrait to landscape mode?

Thanks so much for your amazing work.

PS: Is there a newer tutorial online available, the one included is based on version 4. #StaySafe

**Comment by ivanpopelyshev** [comment]
> Is there a prebuilt function I can use from the start that can make my canvas fill the entire screen and keep the content in place and in the same area of the screen regardless if I turn my, lets say (iphone or ipad) from portrait to landscape mode?

No, its usually exercise to learn `resize` and what is Transform. The problem is that all implementations of that thing we saw were slightly off and so we couldnt add them to pixi and recommend as official :( Its always bucked up somewhere.

**Comment by bigtimebuddy** [comment]
In general Pixi doesn't have any tools for detecting screen ortation, but there are other resources online that can point you to the native APIs to do this.

If you're using the Application class, there is however a built in way to auto-resize your view. It would look something like this:

```js
const app = new PIXI.Application({
  resizeTo: window,
});
document.body.appendChild(app.view);
```
The option `resizeTo` can be any HTML element, like a `<div>` element, or it can be the whole window.  This will automatically set your canvas width and height and adjust the renderer dimensions as well.
